src/main/scala/treadle/vcd/VCD.scala

```diff
@@ -36,15 +36,15 @@ object VCD extends LazyLogging {
   val numberOfIdChars: Int = idChars.length
 
   // A number of regular expressions to parse vcd lines
-  val SectionHeader: Regex = """^\$([^\$]+) *$""".r
+  val SectionHeader: Regex = """^\$([^$]+) *$""".r
   val EndSection:    Regex = """^\$end *$""".r
 
   val ScopedModule: Regex = """\s*(?i)(\S+)\s+(\S+)\s*""".r
   val JustScoped:   Regex = """\s*(\S+)\s*""".r
 
   val VarSpec:            Regex = """\s*(\w+)\s+(\d+)\s+(\S+)\s+([\S ]+)\s*""".r
   val ValueChangeScalar:  Regex = """\s*(\d+)(\S+)\s*""".r
-  val ValueChangeVector:  Regex = """\s*([rbh])([0-9\.]+)\s*""".r
+  val ValueChangeVector:  Regex = """\s*([rbh])([0-9.]+)\s*""".r
```

Comment:
Is this change really correct? Does `.` lose its special meaning inside character class brackets? It seems logical to do so - otherwise, what would be the difference between `.` and `[.]`?

Reply:
Yes. dot loses it's wildcard-ness when in brackets.